waiter/src/waiter/util/utils.clj

```diff
@@ -554,6 +554,16 @@
   [params flag]
   (Boolean/parseBoolean (str (get params flag "false"))))
 
+(defn parse-int
+  "Returns either the input as an integer or nil if there was an error in parsing."
+  [header-value]
+  (try
+    (when header-value
+      (Integer/parseInt (str header-value)))
+    (catch Exception _
+      (log/warn "cannot convert header to an int:" header-value)
```

Comment:
Why is this a warning-level log, whereas other places that we fail to parse things it just emits an info-level log?

Also, why is the parameter here called `header-value` and the log line also says "header"? Why is this function specific to headers? It seems like there are other places we might want to use this generic-sounding `parse-int` helper.